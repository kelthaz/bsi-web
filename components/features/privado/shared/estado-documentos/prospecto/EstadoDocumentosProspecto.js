/* eslint-disable jsx-a11y/label-has-associated-control */
import PropTypes from 'prop-types';
import { useState } from 'react';
import SvgAprobado from '../../../../../svgs/estados/SvgAprobado';
import SvgPendiente from '../../../../../svgs/estados/SvgPendiente';
import SvgRechazado from '../../../../../svgs/estados/SvgRechazado';

import styles from '../estado-documentos.module.scss';

// ESTADOS: 0: Pendiente, 1: Aprobado, 2: Rechazado
const EstadoDocumentosProspecto = ({ documentos }) => {

  const [documentosState, setDocumentosState] = useState(documentos);

  const iconoEstado = (estado) => {
    switch (estado) {
      case 0:
        return <SvgPendiente />;
      case 1:
        return <SvgAprobado />;
      case 2:
        return <SvgRechazado />;
      default:
        return <></>;
    }
  };

  const uploadFile = () => {
    // TODO: Implementar funciÃ³n para subir archivo al servidor.
  };

  const onChangeFileHandler = (event, index) => {
    if (!event.target.files[0]) {
      return;
    }
    uploadFile();
    documentosState[index].ruta = event.target.files[0].name;
    setDocumentosState([...documentosState]);
  };

  return (
    <div className="w-100">
      {documentosState.map((documento, index) => (
        <div className="row my-3" key={documento.id}>
          <div className="col-4 my-auto"><span className="body2">{documento.nombre}</span></div>
          <div className={`col my-auto ${styles.overme}`}>
            <span className="link" title={documento.ruta}>{documento.ruta}</span>
          </div>
          {documento.estado === 2 &&
            <div className="col-3 my-auto">
              <label htmlFor={`inputFile-${documento.nombre}`} className="btn-mini-secondary">Editar</label>
              <input
                accept="image/png, image/jpg, application/pdf"
                type="file"
                id={`inputFile-${documento.nombre}`}
                onChange={(event) => onChangeFileHandler(event, index)}
                hidden
              />
            </div>
          }
          <div className="col-2 my-auto">{iconoEstado(documento.estado)}</div>
        </div>
      ))}
    </div>
  );
};

EstadoDocumentosProspecto.propTypes = {
  documentos: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default EstadoDocumentosProspecto;
