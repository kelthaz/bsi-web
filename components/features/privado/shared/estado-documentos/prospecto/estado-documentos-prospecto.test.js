import React from 'react';
import { mount } from 'enzyme';
import EstadoDocumentosProspecto from './EstadoDocumentosProspecto';

describe('Pruebas en el componente EstadoDocumentosProspecto', () => {
  it('Debe coincidir con la snapshot', () => {
    // arrange
    const documentos = [
      { id: 1, nombre: 'Identificación oficial (Frente)', ruta: 'ALE_INE.jpg', estado: 1 },
      { id: 2, nombre: 'Identificación oficial (Reverso)', ruta: 'ALE_INE.jpg', estado: 1 },
      { id: 3, nombre: 'Comprobante de domicilio', ruta: 'comprobante_luz_factura_hogar.pdf', estado: 1 },
      { id: 4, nombre: 'Poderes notariales', ruta: 'poderes.pdf', estado: 0 },
      { id: 5, nombre: 'Acta constitutiva', ruta: 'acta.pdf', estado: 2 },
      { id: 6, nombre: 'Reformas' },
      { id: 7, nombre: 'Prueba de vida', ruta: 'prueba.mv', estado: 1 },
      { id: 8, nombre: 'Selfie', ruta: 'selfie.png', estado: 1 },
    ];
    const wrap = mount(<EstadoDocumentosProspecto documentos={documentos} />);
    const fileContents = 'file contents';
    const file = new Blob([fileContents], {type : 'text/plain'});
    const link = {
      addEventListener: jest.fn(),
      readAsDataURL: jest.fn()
    };
    // jest.spyOn(window, 'FileReader').mockImplementation(() => link);
    window.FileReader = jest.fn(() => link);
    // act
    wrap.find('input').first().simulate('change', {target: {files: [file]}});
    wrap.find('input').first().simulate('change', {target: {files: []}});
    // assert
    expect(wrap).toMatchSnapshot();
  });
});
