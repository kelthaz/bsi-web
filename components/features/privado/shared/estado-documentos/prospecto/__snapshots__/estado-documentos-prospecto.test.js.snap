// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente EstadoDocumentosProspecto Debe coincidir con la snapshot 1`] = `
<EstadoDocumentosProspecto
  documentos={
    Array [
      Object {
        "estado": 1,
        "id": 1,
        "nombre": "Identificaci贸n oficial (Frente)",
        "ruta": "ALE_INE.jpg",
      },
      Object {
        "estado": 1,
        "id": 2,
        "nombre": "Identificaci贸n oficial (Reverso)",
        "ruta": "ALE_INE.jpg",
      },
      Object {
        "estado": 1,
        "id": 3,
        "nombre": "Comprobante de domicilio",
        "ruta": "comprobante_luz_factura_hogar.pdf",
      },
      Object {
        "estado": 0,
        "id": 4,
        "nombre": "Poderes notariales",
        "ruta": "poderes.pdf",
      },
      Object {
        "estado": 2,
        "id": 5,
        "nombre": "Acta constitutiva",
        "ruta": undefined,
      },
      Object {
        "id": 6,
        "nombre": "Reformas",
      },
      Object {
        "estado": 1,
        "id": 7,
        "nombre": "Prueba de vida",
        "ruta": "prueba.mv",
      },
      Object {
        "estado": 1,
        "id": 8,
        "nombre": "Selfie",
        "ruta": "selfie.png",
      },
    ]
  }
>
  <div
    className="w-100"
  >
    <div
      className="row my-3"
      key="1"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Identificaci贸n oficial (Frente)
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
          title="ALE_INE.jpg"
        >
          ALE_INE.jpg
        </span>
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgAprobado>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#39C6AC"
              r="9"
            />
            <path
              clipRule="evenodd"
              d="M14.7071 5.29289C15.0976 5.68342 15.0976 6.31658 14.7071 6.70711L7.70711 13.7071C7.31658 14.0976 6.68342 14.0976 6.29289 13.7071L3.29289 10.7071C2.90237 10.3166 2.90237 9.68342 3.29289 9.29289C3.68342 8.90237 4.31658 8.90237 4.70711 9.29289L7 11.5858L13.2929 5.29289C13.6834 4.90237 14.3166 4.90237 14.7071 5.29289Z"
              fill="white"
              fillRule="evenodd"
            />
          </svg>
        </SvgAprobado>
      </div>
    </div>
    <div
      className="row my-3"
      key="2"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Identificaci贸n oficial (Reverso)
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
          title="ALE_INE.jpg"
        >
          ALE_INE.jpg
        </span>
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgAprobado>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#39C6AC"
              r="9"
            />
            <path
              clipRule="evenodd"
              d="M14.7071 5.29289C15.0976 5.68342 15.0976 6.31658 14.7071 6.70711L7.70711 13.7071C7.31658 14.0976 6.68342 14.0976 6.29289 13.7071L3.29289 10.7071C2.90237 10.3166 2.90237 9.68342 3.29289 9.29289C3.68342 8.90237 4.31658 8.90237 4.70711 9.29289L7 11.5858L13.2929 5.29289C13.6834 4.90237 14.3166 4.90237 14.7071 5.29289Z"
              fill="white"
              fillRule="evenodd"
            />
          </svg>
        </SvgAprobado>
      </div>
    </div>
    <div
      className="row my-3"
      key="3"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Comprobante de domicilio
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
          title="comprobante_luz_factura_hogar.pdf"
        >
          comprobante_luz_factura_hogar.pdf
        </span>
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgAprobado>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#39C6AC"
              r="9"
            />
            <path
              clipRule="evenodd"
              d="M14.7071 5.29289C15.0976 5.68342 15.0976 6.31658 14.7071 6.70711L7.70711 13.7071C7.31658 14.0976 6.68342 14.0976 6.29289 13.7071L3.29289 10.7071C2.90237 10.3166 2.90237 9.68342 3.29289 9.29289C3.68342 8.90237 4.31658 8.90237 4.70711 9.29289L7 11.5858L13.2929 5.29289C13.6834 4.90237 14.3166 4.90237 14.7071 5.29289Z"
              fill="white"
              fillRule="evenodd"
            />
          </svg>
        </SvgAprobado>
      </div>
    </div>
    <div
      className="row my-3"
      key="4"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Poderes notariales
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
          title="poderes.pdf"
        >
          poderes.pdf
        </span>
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgPendiente>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#AAAAAA"
              r="9"
            />
          </svg>
        </SvgPendiente>
      </div>
    </div>
    <div
      className="row my-3"
      key="5"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Acta constitutiva
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
        />
      </div>
      <div
        className="col-3 my-auto"
      >
        <label
          className="btn-mini-secondary"
          htmlFor="inputFile-Acta constitutiva"
        >
          Editar
        </label>
        <input
          accept="image/png, image/jpg, application/pdf"
          hidden={true}
          id="inputFile-Acta constitutiva"
          onChange={[Function]}
          type="file"
        />
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgRechazado>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#F3685F"
              r="9"
            />
            <path
              d="M5 5L13 13"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="2"
            />
            <path
              d="M13 5L5 13"
              stroke="white"
              strokeLinecap="round"
              strokeWidth="2"
            />
          </svg>
        </SvgRechazado>
      </div>
    </div>
    <div
      className="row my-3"
      key="6"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Reformas
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
        />
      </div>
      <div
        className="col-2 my-auto"
      />
    </div>
    <div
      className="row my-3"
      key="7"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Prueba de vida
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
          title="prueba.mv"
        >
          prueba.mv
        </span>
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgAprobado>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#39C6AC"
              r="9"
            />
            <path
              clipRule="evenodd"
              d="M14.7071 5.29289C15.0976 5.68342 15.0976 6.31658 14.7071 6.70711L7.70711 13.7071C7.31658 14.0976 6.68342 14.0976 6.29289 13.7071L3.29289 10.7071C2.90237 10.3166 2.90237 9.68342 3.29289 9.29289C3.68342 8.90237 4.31658 8.90237 4.70711 9.29289L7 11.5858L13.2929 5.29289C13.6834 4.90237 14.3166 4.90237 14.7071 5.29289Z"
              fill="white"
              fillRule="evenodd"
            />
          </svg>
        </SvgAprobado>
      </div>
    </div>
    <div
      className="row my-3"
      key="8"
    >
      <div
        className="col-4 my-auto"
      >
        <span
          className="body2"
        >
          Selfie
        </span>
      </div>
      <div
        className="col my-auto overme"
      >
        <span
          className="link"
          title="selfie.png"
        >
          selfie.png
        </span>
      </div>
      <div
        className="col-2 my-auto"
      >
        <SvgAprobado>
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9"
              cy="9"
              fill="#39C6AC"
              r="9"
            />
            <path
              clipRule="evenodd"
              d="M14.7071 5.29289C15.0976 5.68342 15.0976 6.31658 14.7071 6.70711L7.70711 13.7071C7.31658 14.0976 6.68342 14.0976 6.29289 13.7071L3.29289 10.7071C2.90237 10.3166 2.90237 9.68342 3.29289 9.29289C3.68342 8.90237 4.31658 8.90237 4.70711 9.29289L7 11.5858L13.2929 5.29289C13.6834 4.90237 14.3166 4.90237 14.7071 5.29289Z"
              fill="white"
              fillRule="evenodd"
            />
          </svg>
        </SvgAprobado>
      </div>
    </div>
  </div>
</EstadoDocumentosProspecto>
`;
